src/utils/helpers/validatePage.tsx

```diff
@@ -0,0 +1,113 @@
+import {
+  ApplicationElementStates,
+  ElementState,
+  ProgressStatus,
+  ResponsesFullByCode,
+  ResponseFull,
+} from '../../utils/types'
+
+import { TemplateElementCategory } from '../../utils/generated/graphql'
+
+export const PROGRESS_STATUS: { [key: string]: ProgressStatus } = {
+  NOT_VALID: 'NOT_VALID',
+  VALID: 'VALID',
+  INCOMPLETE: 'INCOMPLETE',
+}
+
+export const getCombinedStatus = (
+  array: { status: ProgressStatus }[] | undefined,
```

Comment:
I think it's much simpler to just do `some` here:

pages.some(({status} )=> status === NOT_VALID) return NOT_VALID
pages.some(({status} )=> status === INCOMPLETE) return INCOMPLETE
return VALID

Reply:
Yes! I agree (only saw this after I commented on the previous one).
Change [here](https://github.com/openmsupply/application-manager-web-app/pull/127/commits/c55bfbf478f1f7b772b0ce4a506fe4cbb246e5b8)